airflow/providers/microsoft/azure/hooks/azure_batch.py

```diff
@@ -0,0 +1,348 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import time
+from datetime import timedelta
+from typing import Optional
+
+from azure.batch import BatchServiceClient, batch_auth, models as batch_models
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils import timezone
+
+
+class AzureBatchHook(BaseHook):
+    """
+    Hook for Azure Batch APIs
+    """
+
+    def __init__(self, azure_batch_conn_id='azure_batch_default'):
+        super().__init__()
+        self.conn_id = azure_batch_conn_id
+        self.connection = self.get_conn()
```

Comment:
I would say we can remove `self.connection` from here and use the get_conn or get_connection method in the `execute` method of Operator

Reply:
Alright. I'll fix it. Thanks :)

Reply:
I have tried to add this but I keep getting binascii.Error: Incorrect padding when I run the test for operator